server/activity/activity.model.js

```diff
@@ -166,8 +166,9 @@ class Activity extends Model {
       });
   }
 
-  remove(options = {}) {
-    if (!options.recursive) return this.destroy(options);
+  removeOrRestore(opts = {}) {
+    const { recursive, soft, isRestore } = opts;
+    if (!recursive) return isRestore ? this.restore(opts) : this.destroy(opts);
```

Comment:
Instead having ```isRestore``` flag, could you just check the ```deletedAt``` or am I missing something here? On both ```Activity``` & ```TeachingElement``` models? 🤔

Reply:
I removed `restored` field from database. But I also removed on client `isRestored` flag, so restore icon will render for all removed revision items. Furthermore, I left activity model as it was because now when using `deletedAt` instead of `isRestore` flag in `activity.model` if I click removed revision item for the second time, item will be removed in database. @abasic approved this approach for now, let me know if you reconsider.